pkg/controller/controller.go

```diff
@@ -1124,6 +1129,26 @@ func (p *csiProvisioner) getSnapshotSource(ctx context.Context, claim *v1.Persis
 		}
 	}
 
+	if p.preventVolumeModeConversion {
+		if snapContentObj.Spec.SourceVolumeMode != nil && snapContentObj.Spec.SourceVolumeMode != claim.Spec.VolumeMode {
+			// Attempt to modify volume mode during volume creation.
+			// Verify if this volume is allowed to alter its mode.
+			allowVolumeModeChange, ok := snapContentObj.Annotations[annAllowVolumeModeChange]
```

Comment:
The snapshots created before turning on the prevention flag wont land here , isnt it  ? whats the desired workflow for those @RaunakShah @xing-yang ?

Reply:
That's correct. Those objects would have `snapContentObj.Spec.SourceVolumeMode` as `nil` so there is no change in behaviour for preexisting snapshots

Reply:
@xing-yang is above workflow looks good to you ?

Reply:
Yes